### ISSUE TITLE
[Coverity CID :195871]Integer handling issues in /drivers/counter/counter_ll_stm32_rtc.c

### ISSUE THREAD

**Issue by aasthagr** [zephyrproject-rtos/zephyr#14397]
Static code scan issues seen in File: /drivers/counter/counter_ll_stm32_rtc.c
 Category: Integer handling issues
 Function: rtc_stm32_read
 Component: Drivers
 CID: 195871
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

**Comment by erwango** [comment]
Cf analysis in #14415.
Will be fixed when upgrading SDK to 0.10.0.

**Comment by ceolin** [comment]
@erwango actually coverity spot this problem even with latest sdk. Question, why don't you cast this way:
```
ticks = counter_us_to_ticks(dev, (u64_t)ts * USEC_PER_SEC);
```
Here the we ensure that the multiplication won't overflow

**Comment by ceolin** [comment]
actually you don't need the cast at all, `ts` is `time_t`

**Comment by ceolin** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/14799

**Comment by erwango** [comment]
@ceolin , cf my comment in #14799, none of the proposal works actually.

**Comment by galak** [comment]
Can we close this now?

**Comment by erwango** [comment]
Fixed by #14799
